cmdx/config.go

```diff
@@ -55,19 +64,29 @@ func (c *Config) Init(cfg interface{}) error {
 		os.MkdirAll(configDir("odpf"), 0700)
 	}
 
-	if err := ioutil.WriteFile(c.filename, data, 0655); err != nil {
+	if err := os.WriteFile(c.filename, data, 0655); err != nil {
 		return err
 	}
 	return nil
 }
 
 func (c *Config) Read() (string, error) {
-	cfg, err := ioutil.ReadFile(c.filename)
+	cfg, err := os.ReadFile(c.filename)
 	return string(cfg), err
 }
 
-func (c *Config) Load(cfg interface{}) error {
-	loader := config.NewLoader(config.WithFile(c.filename))
+func (c *Config) Load(cfg interface{}, opts ...ConfigLoaderOpts) error {
```

Comment:
How are we creating `Flags` here? Are they supposed to be initialised manually outside this package?
Can we have a way to iterate over the `cfg` and initialize by using the `cmdx` struct tags as tag names?
We could also try to utilize the default value from the struct as default flag value.

Reply:
@rohilsurana I just remember, the reason why the flags are initialized manually because flags could have various data types (string, int, etc..) auto-creating it would make it more complex.

Reply:
yeah that could be a problem, for now lets leave it and we can have it in future if needed.